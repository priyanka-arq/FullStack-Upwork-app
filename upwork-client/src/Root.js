import React from "react";
import configureStore from "./redux/configureStore";
import { Provider } from "react-redux";

const store = configureStore();

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
