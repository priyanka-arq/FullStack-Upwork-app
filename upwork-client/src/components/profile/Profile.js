import React, { Component } from "react";
import { CloudinaryContext, Transformation, Image } from "cloudinary-react";
import ProfileCard from "./ProfileCard";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import * as profileActions from "../../redux/actions/profileActions";
import Loading from "./Loading";
import styled from "styled-components";

const ProfileDisplay = styled.div`
  display: flex;
  flex-direction: row;
`;

class Profile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      profiles: [],
      isLoading: null,
    };
  }

  componentDidMount() {
    const profiles = this.props.profiles;
    console.log("profiles", profiles);
    this.props.loadProfiles().catch((error) => {
      alert("Loading failed" + error);
    });

    if (profiles.length > 0) {
      const isLoading = false;
      this.setState({ isLoading: isLoading });
    }
  }

  render() {
    return (
      <div>
        {this.props.profiles &&
          this.props.profiles.map((profile) => (
            <ProfileCard
              key={profile._id}
              name={profile.name}
              title={profile.title}
              skills={
                profile.skills
                  ? profile.skills.map((skill) => (
                      <div>
                        <p>{skill}</p>
                      </div>
                    ))
                  : []
              }
              redirect={`profile/${profile._id}`}
            >
              <CloudinaryContext cloudName="didtkbpn7">
                <Image publicId={profile.imageUrl}>
                  <Transformation
                    crop="scale"
                    width="300"
                    height="300"
                    dpr="auto"
                    responsive_placeholder="blank"
                  />
                </Image>
              </CloudinaryContext>
            </ProfileCard>
          ))}
        {!this.props.profiles && <Loading show={this.state.isLoading} />}
      </div>
    );
  }
}

// Profile.prototype = {
//   profiles: PropTypes.array.isRequired,
// };

const mapStateToProps = (state) => {
  return {
    profiles: state.profiles,
  };
};

const mapDispatchToProps = (dispatch) => ({
  loadProfiles: () => dispatch(profileActions.loadProfiles()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
