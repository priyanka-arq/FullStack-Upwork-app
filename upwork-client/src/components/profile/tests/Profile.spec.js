import React from "react";
import Profile from "../Profile";
import Root from "../../../Root";
import { shallow } from "enzyme";
import profiles from "../../../fixtures/profiles";
import toJSON from "enzyme-to-json";

let wrapped, loadProfiles;

beforeEach(() => {
  loadProfiles = jest.fn();
  wrapped = shallow(
    //testing connected version of Profile by wrapping it with <Root />
    <Root>
      <Profile profiles={profiles} loadProfiles={loadProfiles} />
    </Root>
  );
});

test("Should render profile correctly ", () => {
  expect(toJSON(wrapped)).toMatchSnapshot();
});
