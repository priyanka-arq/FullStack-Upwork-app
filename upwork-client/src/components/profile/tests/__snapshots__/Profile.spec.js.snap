// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render profile correctly  1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Profile)
    loadProfiles={[MockFunction]}
    profiles={
      Array [
        Object {
          "aboutme": "I have a great passion about solving real world problems and helping people. And now, with the skills that I have been learning in programming and computer science from the Software Engineering Course at General Assembly, I will be able to do that with new technologies. I believe in the power of technology and its impact to everyone's daily lives.",
          "email": "test@ga.com",
          "github": "https://github.com/little-chi-mai",
          "id": 1,
          "imageUrl": null,
          "linkedIn": "https://linkedin.com/in/chi-mai-nguyenn",
          "name": "Test",
          "phone": "04444444444",
          "portfolio": "https://little-chi-mai.github.io/portfolio",
          "postcode": 2000,
          "previewSource": "",
          "resume": "",
          "skills": Array [
            "React",
            "Node",
            "Java",
            "CSS",
          ],
          "suburb": "Sydney",
          "title": "Junior Developer",
          "user": 1,
        },
        Object {
          "aboutme": "I have a great passion about solving real world problems and helping people. And now, with the skills that I have been learning in programming and computer science from the Software Engineering Course at General Assembly, I will be able to do that with new technologies. I believe in the power of technology and its impact to everyone's daily lives.",
          "email": "test1@ga.com",
          "github": "https://github.com/little-chi-mai",
          "id": 2,
          "imageUrl": null,
          "linkedIn": "https://linkedin.com/in/chi-mai-nguyenn",
          "name": "Test1",
          "phone": "04444444444",
          "portfolio": "https://little-chi-mai.github.io/portfolio",
          "postcode": 2000,
          "previewSource": "",
          "resume": "",
          "skills": Array [
            "HTML",
            "PHP",
            "Java",
            "CSS",
          ],
          "suburb": "Sydney",
          "title": "Developer",
          "user": 2,
        },
        Object {
          "aboutme": "I have a great passion about solving real world problems and helping people. And now, with the skills that I have been learning in programming and computer science from the Software Engineering Course at General Assembly, I will be able to do that with new technologies. I believe in the power of technology and its impact to everyone's daily lives.",
          "email": "test2@ga.com",
          "github": "https://github.com/little-chi-mai",
          "id": 3,
          "imageUrl": null,
          "linkedIn": "https://linkedin.com/in/chi-mai-nguyenn",
          "name": "Test2",
          "phone": "04444444444",
          "portfolio": "https://little-chi-mai.github.io/portfolio",
          "postcode": 2000,
          "previewSource": "",
          "resume": "",
          "skills": Array [
            "SWIFT",
            "PHP",
            "C#",
            "CSS",
          ],
          "suburb": "Sydney",
          "title": " Ssenior Developer",
          "user": 3,
        },
      ]
    }
  />
</Provider>
`;
